frontend/src/modules/AppWrapper.tsx

```diff
@@ -18,53 +18,65 @@ import {
 	InstrumentationPage,
 } from "Src/pages";
 import { RouteProvider } from "./RouteProvider";
+import { ErrorFallback } from "../components/ErrorFallback"
 import NotFound from "Src/components/NotFound";
 
 const App = () => {
 	const { status } = useThemeSwitcher();
-
 	if (status === "loading") {
 		return <CustomSpinner size="large" tip="Loading..." />;
 	}
 
 	return (
 		<BrowserRouter>
-			<RouteProvider>
-				<BaseLayout>
-					<Suspense fallback={<CustomSpinner size="large" tip="Loading..." />}>
+			<ErrorBoundary FallbackComponent={ErrorFallback}>
+				<Suspense fallback={<CustomSpinner size="large" tip="Loading..."/>}>
+					<Route path={"/"}>
 						<Switch>
-							<Route path={ROUTES.SIGN_UP} exact component={Signup} />
-							<Route path={ROUTES.APPLICATION} exact component={ServicesTable} />
-							<Route path={ROUTES.SERVICE_METRICS} exact component={ServiceMetrics} />
-							<Route path={ROUTES.SERVICE_MAP} exact component={ServiceMap} />
-							<Route path={ROUTES.TRACES} exact component={TraceDetail} />
-							<Route path={ROUTES.TRACE_GRAPH} exact component={TraceGraph} />
-							<Route path={ROUTES.SETTINGS} exact component={SettingsPage} />
-							<Route
-								path={ROUTES.INSTRUMENTATION}
-								exact
-								component={InstrumentationPage}
-							/>
-							<Route path={ROUTES.USAGE_EXPLORER} exact component={UsageExplorer} />
-							<Route
-								path="/"
-								exact
-								render={() => {
-									return localStorage.getItem(IS_LOGGED_IN) === "yes" ? (
-										<Redirect to={ROUTES.APPLICATION} />
-									) : (
-										<Redirect to={ROUTES.SIGN_UP} />
-									);
-								}}
-							/>
-
-							<Route path="*" exact component={NotFound} />
+							<RouteProvider>
+								<BaseLayout>
```

Comment:
Can you also put a error boundary inside of BaseLayout so that if any error occurs it doesn't crash the navigation. Sidebar and navbar should remain functioning.